instrumentation/influxdb-2.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/influxdb/v2_4/InfluxDbClientTest.java

```diff
@@ -0,0 +1,311 @@
+                                String.format("DROP DATABASE \"%s\"", dbName), "DROP", dbName))));
+  }
+
+  @Test
+  void testQueryWithTwoArguments() {
+    Query query = new Query("SELECT * FROM cpu_load where test1 = 'influxDb'", databaseName);
+    influxDb.query(query, TimeUnit.MILLISECONDS);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("SELECT " + databaseName)
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfying(
+                            attributeAssertions(
+                                "SELECT * FROM cpu_load where test1 = ?",
+                                "SELECT",
+                                databaseName))));
+  }
+
+  @Test
+  void testQueryWithThreeArguments() throws InterruptedException {
+    Query query =
+        new Query(
+            "SELECT * FROM cpu_load where time >= '2022-01-01T08:00:00Z' AND time <= '2022-01-01T20:00:00Z'",
+            databaseName);
+    BlockingQueue<QueryResult> queue = new LinkedBlockingQueue<>();
+
+    influxDb.query(query, 2, result -> queue.add(result));
+    Thread.sleep(2000);
+    queue.poll(20, TimeUnit.SECONDS);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("SELECT " + databaseName)
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfying(
+                            attributeAssertions(
+                                "SELECT * FROM cpu_load where time >= ? AND time <= ?",
+                                "SELECT",
+                                databaseName))));
+  }
+
+  @Test
+  void testQueryWithThreeArgumentsCallback() throws InterruptedException {
+    Query query = new Query("SELECT * FROM cpu_load", databaseName);
+    BlockingQueue<QueryResult> queue = new LinkedBlockingQueue<>();
+
+    influxDb.query(query, 2, result -> queue.add(result));
+    Thread.sleep(2000);
+    queue.poll(20, TimeUnit.SECONDS);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("SELECT " + databaseName)
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfying(
+                            attributeAssertions(
+                                "SELECT * FROM cpu_load", "SELECT", databaseName))));
+  }
+
+  @Test
+  void testQueryWithFiveArguments() throws InterruptedException {
+    CountDownLatch countDownLatch = new CountDownLatch(1);
+    CountDownLatch countDownLatchFailure = new CountDownLatch(1);
+    Query query =
+        new Query(
+            "SELECT MEAN(water_level) FROM h2o_feet where time = '2022-01-01T08:00:00Z'; SELECT water_level FROM h2o_feet LIMIT 2",
+            databaseName);
+    influxDb.query(
```

Comment:
One thing we test for apis that have that kind of callbacks is that the callback also runs with expected context. To do this we add a span named `parent` around the operation we are going to test and create a span named `child` in the callback. Span from callback should be a sibling for the span for the executed operation. See https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/42ba962aa89b8401513b1a92d4e9b246da7b8dbd/instrumentation/okhttp/okhttp-3.0/testing/src/main/java/io/opentelemetry/instrumentation/okhttp/v3_0/AbstractOkHttp3Test.java#L148 for an example, you can find more samples by searching for `"child"`.

Reply:
> One thing we test for apis that have that kind of callbacks is that the callback also runs with expected context. To do this we add a span named `parent` around the operation we are going to test and create a span named `child` in the callback. Span from callback should be a sibling for the span for the executed operation. See
> 
> https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/42ba962aa89b8401513b1a92d4e9b246da7b8dbd/instrumentation/okhttp/okhttp-3.0/testing/src/main/java/io/opentelemetry/instrumentation/okhttp/v3_0/AbstractOkHttp3Test.java#L148
> 
> for an example, you can find more samples by searching for `"child"`.

I added relevant callback tests. But I found because of the implement of callback logic in  https://github.com/influxdata/influxdb-java/blob/0c690c8d8a6627f525e05d68e0571e089c61845e/src/main/java/org/influxdb/impl/InfluxDBImpl.java#L637 the `child` span is the child of the executed operation instead of sibling:(

Reply:
I implemented running callbacks in the scope of the parent span.